### PR TITLE
Initial BSP layer definitions

### PR DESCRIPTION
**Pull request by jpetrina**
Pushing initial layer work, with `meta-switchbsp` as platform name, for review.

Thanks!

cc @damirsamardzic

### CODE REVIEW

**Review by taraschornyiplv — APPROVED** [review]

File: conf/machine/arm64-all.conf [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require conf/machine/include/common.inc
+require conf/machine/include/arm/arch-armv8a.inc
+
+MACHINE_FEATURES = "pci serial"
+
+KERNEL_DEVICETREE = " \
+    freescale/fsl-ls1043a-rdb.dtb \ 
+    freescale/fsl-ls1046a-rdb-sdk.dtb \
+    freescale/fsl-ls2080a-rdb.dtb \
```
Comment (sonoble) [inline conf/machine/arm64-all.conf]: Is there a developer/customer need for these NXP RDBs? For simplicity they can probably be removed along with any patches unless there is a specific RDB somewhere that is being used for testing.  I believe only the ls1043a-rdb has been used previously and specifically for initial porting work.

Comment (damirsamardzic) [reply conf/machine/arm64-all.conf]: If these are not needed, we can remove them. Please confirm.

File: conf/layer.conf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+LAYERSERIES_COMPAT_switchbsp = "styhead"
+
+BBFILES_DYNAMIC += " \
+    openembedded-layer:${LAYERDIR}/dynamic-layers/openembedded-layer/*/*/*.bb \
+    openembedded-layer:${LAYERDIR}/dynamic-layers/openembedded-layer/*/*/*.bbappend \
+    networking-layer:${LAYERDIR}/dynamic-layers/networking-layer/*/*/*.bb \
+    networking-layer:${LAYERDIR}/dynamic-layers/networking-layer/*/*/*.bbappend \
+    meta-linux-mainline:${LAYERDIR}/dynamic-layers/meta-linux-mainline/*/*/*.bb \
+    meta-linux-mainline:${LAYERDIR}/dynamic-layers/meta-linux-mainline/*/*/*.bbappend \
+"
```
Comment (KanjiMonster) [inline conf/layer.conf]: These paths (`${LAYERDIR}/dynamic-layers/...`) do not exist, so the `BBFILES_DYNAMIC` definition does not make any sense.

File: conf/machine/arm64-all.conf (hunk 1 above)
Comment (jpetrina) [reply conf/machine/arm64-all.conf]: @sonoble, I went ahead and removed these, we can bring them back if they are needed, thanks.

File: conf/layer.conf (hunk 2 above)
Comment (jpetrina) [reply conf/layer.conf]: @KanjiMonster, I removed this leftover boilerplate code, thanks!

File: conf/layer.conf [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+# We have a packages directory, add to BBFILES
+BBFILES += "${LAYERDIR}/recipes-*/*/*.bb \
+            ${LAYERDIR}/recipes-*/*/*.bbappend"
+
+BBFILE_COLLECTIONS += "switchbsp"
+BBFILE_PATTERN_switchbsp := "^${LAYERDIR}/"
+BBFILE_PRIORITY_switchbsp = "5"
+
+LAYERDEPENDS_switchbsp = "core"
```
Comment (KanjiMonster) [inline conf/layer.conf]: Your README.md says this layer depends meta-oe, but your LAYERDEPENDS does not:

```suggestion
LAYERDEPENDS_switchbsp = "core"
LAYERDEPENDS_switchbsp += "openembedded-layer"
```

Comment (jpetrina) [reply conf/layer.conf]: Thanks, added. I was surprised to see other 3rd party layers not enforcing this dep.

**Pull request merged**